tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,306 @@
+
+RHEL_7_REPO = 'rhel-*-optional-rpms rhel-*-extras-rpms rhel-ha-for-rhel-*-server-rpms'
+RHEL_7_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm']
+
+RHEL_8_REPO = 'codeready-builder-for-rhel-8-$(arch)-rpms'
+RHEL_8_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm']
+
+RHEL_9_REPO = 'codeready-builder-for-rhel-9-$(arch)-rpms'
+RHEL_9_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm']
+
+
+class EPEL(ToggleableFeature):
+    KEY = 'epel'
+
+    def enable(self) -> None:
+        distro = self.guest_distro
+        sudo = self.guest_sudo
+
+        if distro == Distro.FEDORA:
+            self.info('Nothing to do on Fedora for EPEL')
+        elif distro == Distro.CENTOS_7:
+            # yum install epel-release
+            self.info('Enable EPEL on CentOS 7')
+            self.guest.execute(
+                ShellScript(f'{sudo} yum -y install {" ".join(CENTOS_7_PACKAGES)}'),
+                silent=True)
+        elif distro == Distro.CENTOS_STREAM_8:
+            # dnf config-manager --set-enabled powertools
+            # dnf -y install epel-release epel-next-release
+            self.info('Enable EPEL on CentOS Stream 8')
+            self.guest.execute(
+                ShellScript(f'{sudo} dnf config-manager --set-enabled {FEDORA_REPO}')
+                & ShellScript(f'{sudo} dnf -y install {" ".join(FEDORA_PACKAGES)}'),
+                silent=True)
+        elif distro == Distro.CENTOS_STREAM_9:
+            # dnf config-manager --set-enabled crb
+            # dnf -y install epel-release epel-next-release
+            self.info('Enable EPEL on CentOS Stream 9')
+            self.guest.execute(
+                ShellScript(f'{sudo} dnf config-manager --set-enabled {CENTOS_STREAM_9_REPO}')
+                & ShellScript(f'{sudo} dnf -y install {" ".join(FEDORA_PACKAGES)}'),
+                silent=True)
+        elif distro == Distro.RHEL_7:
+            # subscription-manager repos --enable rhel-*-optional-rpms \
+            #               --enable rhel-*-extras-rpms \
+            #               --enable rhel-ha-for-rhel-*-server-rpms
+            # yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+            self.info('Enable EPEL on RHEL 7')
+            self.guest.execute(
+                ShellScript(f'{sudo} subscription-manager repos --enable {RHEL_7_REPO}')
+                + ShellScript(f'{sudo} yum -y install {" ".join(RHEL_7_PACKAGES)}'),
+                silent=True)
+        elif distro == Distro.RHEL_8:
+            # subscription-manager repos --enable codeready-builder-for-rhel-8-$(arch)-rpms
+            # dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+            self.info('Enable EPEL on RHEL 8')
+            self.guest.execute(
+                ShellScript(f'{sudo} subscription-manager repos --enable {RHEL_8_REPO}')
```

Comment:
There is an expectation that a test system is subscribed.. This is in general a false assumption.

Reply:
`subscription-manager repos ...` lines are removed.